### PR TITLE
Support  containerized SDK

### PR DESCRIPTION
**Pull request by masami256**
# Purpose of pull request

This PR supports building containerized SDK. To do it, we needed umoci and skopeo packages to create containerized SDK. To setup cross build environment values such as TARGET_PREFIX,. CC,  and so on,  we use the mechanism of /etc/profile.d . Install environment setup script into the /etc/profile.d directory to execute setup script when user logged into  container image.

Above steps is enough for using a containerized SDK. However, this environment doesn't contain user other than root. This will make problem when developer use volume mount to share a directory both host and container. If developer creates a file from container, this file owner will be root on the host. It's not expected behavior for building software from container. So, we use containerized SDK as base image instead of using image directory.  We provide simple Dockerfile  and docker-compose.yml file to create a user which uid is same as host user's uid to resolve uid mapping problem.

# Test
## How to test

1. Add following setting in the conf/local.conf

```
BB_ENV_PASSTHROUGH_ADDITIONS="$BB_ENV_EXTRAWHITE SDK_FORMATS"
SDK_FORMATS="docker-archive"
```
2. Build sdk with 'bitbake emlinux-image-base -c populate_sdk'
3. exit from container if you executed run.sh to enter build env
4. move directory to image deploy dir.  e.g. /your build directory/tmp/deploy/images/qemu-arm64/
5. load  containerized SDK.  sdk image name has .docker-archive extension. e.g. emlinux-image-base-sdk-emlinux-bookworm-arm64.docker-archive
```
docker load -i <containerized sdk image name>
```
6. build build environment image from  containerized SDK
```
docker compose build
```
7. logged into container
```
docker compose run --rm emlinux3-sdk
```
## Test result

```
$ docker image load -i emlinux-image-base-sdk-emlinux-bookworm-arm64.docker-archive
f48b2f3a238f: Loading layer [==================================================>]  823.5MB/823.5MB
Loaded image: emlinux-image-base-sdk-emlinux-bookworm-arm64:1.0-r0
$ docker compose build
>>> snip <<<
$ id -u
1000
$ docker compose run --rm emlinux3-sdk
>>> snip <<<
sdkuser@0821337e7ae0:/$ id -u
1000
```

### CODE REVIEW

File: templates/containerized-sdk/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        ftp_proxy: $ftp_proxy
+        no_proxy: $no_proxy
+    environment:
+      - http_proxy=$http_proxy
+      - https_proxy=$https_proxy
+      - ftp_proxy=$ftp_proxy
+      - no_proxy=$no_proxy
+    volumes:
+      - /proc:/proc:rw
+      - /dev/shm:/dev/shm:rw
```
Comment (hiraku-wfs) [inline templates/containerized-sdk/docker-compose.yml:20]: Can we mount the host OS's home directory to share files between the host OS and the container?

e.g.
```
      - ${HOME}:/home-host:rw
```

**Comment by hiraku-wfs** [comment]
I think the `emlinux-image-base-sdk-emlinux-bookworm-arm64.docker-archive` will be generated in `build/tmp/deploy/images/$MACHINE/` directory. So could you add the path information to the description of the PR (or just add `cd ...` before `docker image load`)?

File: templates/containerized-sdk/docker-compose.yml (hunk 1 above)
Comment (masami256) [reply templates/containerized-sdk/docker-compose.yml:20]: Added  following line to share user's $HOME with container.

```
- $HOME:/host-home:rw
```

**Comment by masami256** [comment]
> I think the `emlinux-image-base-sdk-emlinux-bookworm-arm64.docker-archive` will be generated in `build/tmp/deploy/images/$MACHINE/` directory. So could you add the path information to the description of the PR (or just add `cd ...` before `docker image load`)?

Updated.

File: templates/containerized-sdk/docker-compose.yml (hunk 1 above)
Comment (hiraku-wfs) [reply templates/containerized-sdk/docker-compose.yml:20]: Thanks!

**Comment by hiraku-wfs** [comment]
LGTM.

**Comment by hiraku-wfs** [comment]
Thanks!

**Pull request merged**